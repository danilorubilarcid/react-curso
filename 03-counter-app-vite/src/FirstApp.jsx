import PropTypes from  'prop-types';
const newMessage = 'Danilo';


const saludar = function(nombre){
    return `Hola, ${nombre}`;
}


export const PrimerComponente = ({title, subtitle, name}) => {
  
  
    return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subtitle}</p>
    <p>{name}</p>
    </>
  )
}


PrimerComponente.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

PrimerComponente.defaultProps = {
    subtitle: 'subtitulo por defecto'
}