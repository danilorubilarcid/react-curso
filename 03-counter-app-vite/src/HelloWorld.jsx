/* export function App(){
    return (<h1>Hola mundo!</h1>);
}; */


import React from 'react'

export const HelloWorld = () => {
  return (
    <h1>Hola Mundo</h1>
  )
}