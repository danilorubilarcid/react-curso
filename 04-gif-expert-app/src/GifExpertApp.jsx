import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['one punch', 'dragon ball']);

    const onAddCategory = (newCategory) => {
        //Mi forma
        //setCategories(categories.concat(['Iracing']));
        //copiando arreglo
        //setCategories([...categories, 'Iracing']);
        //setCategories(cat => [...cat, 'Iracing'])
        setCategories([newCategory, ...categories]);
    }


    return(
        <>
        <h1>GifExpertApp</h1>

        <AddCategory 
        //setCategories={setCategories}
            onNewCategory={ (event) => onAddCategory(event)}
        />

        <ol>
            {
                categories.map( category => {
                    return <li key={category}>{category}</li>
                })
            }


        </ol>


        </>
    );
}
