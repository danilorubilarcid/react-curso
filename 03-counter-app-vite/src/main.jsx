import React from 'react';
import ReactDOM from 'react-dom/client';
import { PrimerComponente } from './FirstApp';
import { HelloWorld } from './HelloWorld';
import './styles.css';
import { CounterApp } from './CounterApp';


ReactDOM.createRoot (document.getElementById('root')).render(
<React.StrictMode>
    <CounterApp value={7}/>
 </React.StrictMode>

);