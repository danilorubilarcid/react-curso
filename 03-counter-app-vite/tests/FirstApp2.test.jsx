import { render, screen } from "@testing-library/react";
import { PrimerComponente } from "../src/FirstApp";

describe('pruebas en first app component', () => {

        const titulo = 'Hola soy el titulo';
        const subtitulo = 'Hola soy el subtitulo';

        test('debe hacer match con el snapshot', () => { 

                const {container} = render(<PrimerComponente title={titulo} subtitle={subtitulo}/>);

                expect(container).toMatchSnapshot();
         })

         test('debe ded mostrar el titulo hola soy el titulo', () => { 

                render(<PrimerComponente title={titulo} subtitle={subtitulo}/>);
                expect(screen.getByText(titulo)).toBeTruthy();
          })

        test('el titulo debe estar en un h1', () => { 
                render(<PrimerComponente title={titulo} subtitle={subtitulo}/>);
                expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(titulo);
         })

         test('debe mostrar el subtitulo enviado por props', () => { 
                render(<PrimerComponente title={titulo} subtitle={subtitulo}/>);
                expect(screen.getByText(subtitulo).innerHTML).toBe(subtitulo);

          })
});
