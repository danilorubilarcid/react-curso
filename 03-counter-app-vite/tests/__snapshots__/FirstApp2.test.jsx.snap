// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en first app component debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy el titulo
  </h1>
  <p>
    Hola soy el subtitulo
  </p>
  <p />
</div>
`;
